app/code/Magento/InventoryBundleIndexer/Indexer/SourceItem/GetAllBundleChildrenSourceItemsIdsWithSku.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryBundleIndexer\Indexer\SourceItem;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product\Type as ProductType;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Model\ResourceModel\SourceItem;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+
+/**
+ * Get only bundle children source items ids.
+ */
+class GetAllBundleChildrenSourceItemsIdsWithSku
```

Comment:
Why do we have a lot of different logic in 
 `\Magento\InventoryBundleIndexer\Indexer\SourceItem\GetAllBundleChildrenSourceItemsIdsWithSku`
and `\Magento\InventoryBundleIndexer\Indexer\SourceItem\GetBundleChildrenSourceItemsIdsWithSku`

Difference is only in `$sourceItemsIds` parameter
So `GetBundleChildrenSourceItemsIdsWithSku` is `GetAllBundleChildrenSourceItemsIdsWithSku` with one more condition

Reply:
Fixed.